src/download.c

```diff
@@ -581,6 +582,11 @@ int swupd_curl_parallel_download_end(void *handle, int *num_downloads)
 		}
 	}
 
+	// If there are download failures at this point return an error
+	if (h->failed != NULL) {
+		ret = -EFULLDOWNLOAD;
```

Comment:
As you are changing this, can you rename this constant? This name doesn't make sense anymore. And can you document the new return code in the function header?

Reply:
ok, I'll rename it as EFILEDOWNLOAD, I think it will be more useful that way